RecentActivity/src/org/sleuthkit/autopsy/recentactivity/ExtractRecycleBin.java

```diff
@@ -0,0 +1,552 @@
+       return artifact.getAttribute(new BlackboardAttribute.Type(BlackboardAttribute.ATTRIBUTE_TYPE.fromID(type.getTypeID())));
+    }
+    
+    private void createRecycleBinArtifactType() throws TskCoreException {
+         try {
+            tskCase.addBlackboardArtifactType(RECYCLE_BIN_ARTIFACT_NAME, "Recycle Bin"); //NON-NLS
+        } catch (TskDataException ex) {
+            logger.log(Level.INFO, String.format("%s may have already been defined for this case", RECYCLE_BIN_ARTIFACT_NAME));
+        }
+
+    }
+    
+    /**
+     * Create the new artifact for the give rFile
+     * 
+     * @param rFile AbstractFile to create the artifact for
+     * @param type Type of artifact to create
+     * @param fileName The original path of the deleted file
+     * @param userName The name of the user that deleted the file
+     * @param dateTime The time in epoch seconds that the file was deleted
+     * 
+     * @return Newly created artifact
+     * 
+     * @throws TskCoreException 
+     */
+    private BlackboardArtifact createArtifact(AbstractFile rFile, BlackboardArtifact.Type type, String fileName, String userName, long dateTime) throws TskCoreException{
+        BlackboardArtifact bba = rFile.newArtifact(type.getTypeID());
+        bba.addAttribute(new BlackboardAttribute(TSK_PATH, getName(), fileName));
+        bba.addAttribute(new BlackboardAttribute(TSK_DATETIME_DELETED, getName(), dateTime));
+        bba.addAttribute(new BlackboardAttribute(TSK_USER_NAME, getName(), userName == null || userName.isEmpty() ? "" : userName));
+        
+        return bba;
+    }
+    
+    private AbstractFile getOrMakeFolder(SleuthkitCase skCase, FsContent dataSource, String path) throws TskCoreException {
+        
+        String parentPath = getParentPath(path);
+        String folderName = getFileName(path);
+        
+        List<AbstractFile> files = null;
+        if(parentPath != null) {
+            if(!parentPath.equals("/")) {
+                parentPath = parentPath + "/";
+            }
+            files = skCase.findAllFilesWhere(String.format("fs_obj_id=%s AND parent_path='%s' AND name='%s'", dataSource.getFileSystemId(), parentPath, folderName != null ? folderName : ""));
+        } else {
+            files = skCase.findAllFilesWhere(String.format("fs_obj_id=%s AND parent_path='/' AND name=''", dataSource.getFileSystemId()));
+        }
+        
+        if(files == null || files.isEmpty()) {
+            AbstractFile parent = getOrMakeFolder(skCase, dataSource, parentPath); 
+            return skCase.addVirtualDirectory(parent.getId(), folderName);
+           
+        } else {
+            return files.get(0);
+        }
+    }
+    
+    private void addFileToDirectory (SleuthkitCase skCase, AbstractFile rFile, Content parentDir, String fileName, long deletedTime) throws TskCoreException {
+        FsContent fsContent = (FsContent)rFile;
```

Comment:
We shoudl either test before casting or simply make the argument be of type FsContent.

Reply:
Agreed, I have already changed this.